### ISSUE TITLE
`snapshot fix remove-files` does not match any files

### ISSUE THREAD

**Issue by Araeos** [kopia/kopia#3663]
I've taken snapshots of a folder `foo` that contains a cache sub-folder under `foo/bar/cache` and want to delete that cache from all snapshots using `snapshot fix remove-files --filename=foo/bar/cache`. These names are just examples and I have for full disclosure not created that exact file structure, but tested various patterns that all do not work as described below:

Expected behavior:
Deletes only the cache folder in `foo/bar`.

What actually happens:
It does not match anything.

I tried various alternatives like
```
snapshot fix remove-files --filename=foo/bar/cache/  # Matches nothing
snapshot fix remove-files --filename=bar/cache       # Matches nothing
snapshot fix remove-files --filename=cache/          # Matches nothing
snapshot fix remove-files --filename=cache           # Deletes all folders and files called cache, many of which I want to keep.
```
None work like I expect or want them to.

**Comment by kien-truong** [comment]
Yeah, that's because currently remove files only match the name component instead of an entire path.

https://github.com/kopia/kopia/blob/f1c141f8be1dc8b554cde67d30469f893e6c86b2/cli/command_snapshot_fix_remove_files.go#L39-L40

It'd be much better if we can remove files using an exclusion file like `.kopiaignore`.

**Comment by felixhauptmann** [comment]
Something like this should work if I understand Kopia correctly, right?

```bash
kopia snapshot list --all "user@host:/path/to/file" \
| grep --only-matching -E "(Ix|k)?[0-9a-f]{32}" \
| xargs -I{} echo "--object-id={}" \
| xargs kopia snapshot fix remove-files --source "user@host"
```

- list all snapshots of the specified file
- filter out the object IDs (`Ix` for large files and `k` for directory listings)
- pass all those object IDs to `snapshot fix remove-files`

Please someone confirm that this makes sense.
